server/lib/deriva/provider.py

```diff
@@ -0,0 +1,26 @@
+from ..data_map import DataMap
+from ..entity import Entity
+from ..bdbag.bdbag_provider import BDBagProvider
+
+from girder.plugins.oauth.providers.globus import Globus
+
+Globus.addScopes(['https://auth.globus.org/scopes/a77ee64a-fb7f-11e5-810e-8c705ad34f60/deriva_all'])
+
+class DerivaProvider(BDBagProvider):
+    def __init__(self) -> None:
+        super().__init__('DERIVA')
+
+    def matches(self, entity: Entity) -> bool:
+        try:
+            return entity.getValue().startswith('https://pbcconsortium.s3.amazonaws.com/')
+        except:
+            return False
+
+    def lookup(self, entity: Entity) -> DataMap:
+        sz = -1
+        if 'size' in entity:
+            sz = entity['size']
+        name = None
```

Comment:
Can we default to something nice like:

```
name = pathlib.Path(urlparse(entity.getValue()).path).name  # where urlparse is from urllib.parse
```

potentially with:

```
if name.endswith(".zip"):
    name = name[:-4]
```

Reply:
We probably should.